torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -591,9 +575,78 @@ def _update_trainable(self) -> None:
         # Create the optim which will work on the param shard
         if not hasattr(self, "optim"):
             self._clear_cache()
-            self._default_device = list(self._per_device_params.keys())[0]
             self.optim = self._optim_constructor(self.partition_parameters()[self.rank], **self._optim_defaults)
             self._sync_param_groups(self.optim.param_groups, self.param_groups)
 
         if self.parameters_as_bucket_view:
             self._setup_flat_buffers()
+
+    def _verify_params(self, params: Any) -> None:
```

Comment:
nit: `_verify_and_init_params`?

Reply:
Good suggestion. Since the function has an important side-effect (initializing `self._all_params`), this function name makes much more sense. Fixed.